crates/walrus-service/src/node/events/event_processor.rs

```diff
@@ -1086,6 +1103,19 @@ impl EventProcessor {
             .collect();
         (first_event, relevant_events)
     }
+
+    /// Updates the cached checkpoint sequence number.
+    fn update_checkpoint_cache(&self, sequence_number: u64) {
+        // Use fetch_max to atomically update the value if sequence_number is greater
+        self.latest_checkpoint_seq_cache
+            .fetch_max(sequence_number, Ordering::SeqCst);
+    }
+
+    /// Gets the latest checkpoint sequence number, preferring the cache.
```

Comment:
This comment seems out of date or maybe just say:  `Gets the latest checkpoint sequence number from the cache`